interpreter/src/interpreter/mod.rs

```diff
@@ -485,20 +487,54 @@ fn call_external_func(ctx: &Context, func: &Function, args: &[GenericValue]) ->
             .find_map(|lib| unsafe { lib.get(name.as_bytes()) }.ok())
     }
 
-    let mut args_ty = Vec::with_capacity(100);
-    let mut new_args = Vec::with_capacity(100);
+    let mut args_ty = Vec::with_capacity(args.len()+8);
+    let mut new_args = Vec::with_capacity(args.len()+8);
     let mut args: Vec<GenericValue> = args.to_vec();
-    for arg in &mut args {
-        match arg {
-            GenericValue::Int32(ref mut i) => {
-                args_ty.push(unsafe { &mut libffi::low::types::sint32 as *mut _ });
-                new_args.push(i as *mut _ as *mut c_void)
+    #[cfg(all(target_arch = "aarch64", target_vendor = "apple"))]
+    {
+        let mut dummy = vec![0i64];
+        for arg in &mut args {
+            if func.is_var_arg() && new_args.len() == 1 {
+                for _ in 0..7 {
+                    args_ty.push(unsafe { &mut libffi::low::types::sint64 as *mut _ });
+                    new_args.push(&mut dummy as *mut _ as *mut c_void);
+                }
+            }
```

Comment:
By using `libffi::low::prep_cif_var`, we don't have to treat macOS(ARM) as a special case.